nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetTCP.java

```diff
@@ -0,0 +1,668 @@
+            .defaultValue("10")
+            .build();
+
+
+    public static final PropertyDescriptor RECEIVE_BUFFER_SIZE = new PropertyDescriptor
+            .Builder().name("Receive Buffer Size")
+            .description("The size of the buffer to receive data in")
+            .required(false)
+            .defaultValue("2048")
+            .addValidator(StandardValidators.createLongValidator(1, 2048, true))
+            .build();
+
+    public static final PropertyDescriptor KEEP_ALIVE = new PropertyDescriptor
+            .Builder().name("Keep Alive")
+            .description("This determines if TCP keep alive is used.")
+            .required(false)
+            .defaultValue("true")
+            .allowableValues("true", "false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("Success")
+            .description("The relationship that all sucessful messages from the WebSocket will be sent to")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("Failure")
+            .description("The relationship that all failed messages from the WebSocket will be sent to")
+            .build();
+
+    private final static List<PropertyDescriptor> propertyDescriptors;
+    private final static Set<Relationship> relationships;
+    private final static Charset charset = Charset.forName(StandardCharsets.UTF_8.name());
+    private final Map<String, String> dynamicAttributes = new HashMap<>();
+    private volatile Set<String> dynamicPropertyNames = new HashSet<>();
+
+    private AtomicBoolean connectedToBackup = new AtomicBoolean();
+
+
+    /*
+    * Will ensure that the list of property descriptors is build only once.
+    * Will also create a Set of relationships
+    */
+    static {
+        List<PropertyDescriptor> _propertyDescriptors = new ArrayList<>();
+        _propertyDescriptors.add(ENDPOINT_LIST);
+        _propertyDescriptors.add(FAILOVER_ENDPOINT);
+        _propertyDescriptors.add(CONNECTION_TIMEOUT);
+        _propertyDescriptors.add(CONNECTION_ATTEMPT_COUNT);
+        _propertyDescriptors.add(BATCH_SIZE);
+        _propertyDescriptors.add(RECEIVE_BUFFER_SIZE);
+        _propertyDescriptors.add(KEEP_ALIVE);
+
+
+        propertyDescriptors = Collections.unmodifiableList(_propertyDescriptors);
+
+        Set<Relationship> _relationships = new HashSet<>();
+        _relationships.add(REL_SUCCESS);
+        _relationships.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(_relationships);
+    }
+
+    private Map<SocketRecveiverThread, Future> socketToFuture = new HashMap<>();
+    private ExecutorService executorService;
+    private ComponentLog log = getLogger();
+    private String originalServerAddressList;
+    private String backupServer;
+    private int batchSize;
```

Comment:
No, since _onTrigger()_ can be executed by a different thread. Basically this is not about synchronization, but about visibility of of values across threads. Basically it is possible that the values set by the _onSchedule_  thread are not visible to _onTrigger()_ thread.